vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/filters/goaway.go

```diff
@@ -56,13 +56,25 @@ func WithProbabilisticGoaway(inner http.Handler, chance float64) http.Handler {
 	}
 }
 
+// WithTerminationGoaway sends a GOAWAY when isTerminating is returning true.
+func WithTerminationGoaway(inner http.Handler, isTerminating func() bool) http.Handler {
+	return &goaway{
+		handler: inner,
+		decider: GoawayDeciderFunc(
+			func(r *http.Request) bool {
+				return isTerminating()
```

Comment:
maybe we should log these requests? they would end up in the new termination.log file.

Reply:
we log all on termination anyway.